storagenode/gracefulexit/worker.go

```diff
@@ -0,0 +1,55 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package gracefulexit
+
+import (
+	"context"
+	"time"
+
+	"github.com/zeebo/errs"
+	"go.uber.org/zap"
+
+	"storj.io/storj/pkg/storj"
+	"storj.io/storj/storagenode/satellites"
+)
+
+// Worker is responsible for completing the graceful exit for a given satellite.
+type Worker struct {
+	log         *zap.Logger
+	satelliteID storj.NodeID
+	satelliteDB satellites.DB
+}
+
+// NewWorker instantiates Worker.
+func NewWorker(log *zap.Logger, satelliteDB satellites.DB, satelliteID storj.NodeID) *Worker {
+	return &Worker{
+		log:         log,
+		satelliteID: satelliteID,
+		satelliteDB: satelliteDB,
+	}
+}
+
+// Run calls the satellite endpoint, transfers pieces, validates, and responds with success or failure.
+// It also marks the satellite finished once all the pieces have been transferred
+func (worker *Worker) Run(ctx context.Context, satelliteID storj.NodeID, done func()) (err error) {
+	defer mon.Task()(&ctx)(&err)
+	defer done()
+	worker.log.Debug("running worker")
+
+	// TODO actually process the order limits
+	// https://storjlabs.atlassian.net/browse/V3-2613
+
+	err = worker.satelliteDB.CompleteGracefulExit(ctx, satelliteID, time.Now().UTC(), satellites.ExitSucceeded, []byte{})
```

Comment:
Why time needs to be `UTC`?

Reply:
I thought we were always using UTC. Is that incorrect?

Reply:
We were using UTC in many places but I think we were removing `.UTC()` from many places later. I don't remember exactly. I don't remember also who was doing that to ask about it... I found something like this https://github.com/storj/storj/pull/2517 but I'm not sure if this applies here. If don't see any issue with this leave it as is.

Reply:
Handle UTC at database level, because otherwise it's easy to forget in services the UTC conversion and it will end up replicating the conversion in more places.

Reply:
Good points.  I have just been in the habit of always using UTC. I've removed here. It is handled in the DB layer.

Reply:
oops I think I've been a purveyor of always UTC, good to know!